### DISCUSSION TITLE
Emote Steal Command - Help

### DISCUSSION THREAD

**Discussion by samjesus8** [DSharpPlus/DSharpPlus#1502] [Support] [answered]
# Summary

I have this Steal Slash Command which basically should allow users to put in an emoji and it will convert it into a GIF or PNG depending on if it is an animated or static emoji.

The command is not finished yet because I am trying to figure out how to get emojis from unknown servers that may be private or the bot might not be in those servers

# My current code

```
        [SlashCommand("steal", "Steal an emote")]
        public async Task EmoteSteal(InteractionContext ctx, [Option("emotename", "Name of the emote (Example - :thumbsup:)")] string emoteName)
        {
            await ctx.DeferAsync();
            string[] emoteArray = emoteName.Split('<', ':', '>');

            try
            {
                var name = ":" + emoteArray[2] + ":";
                var emote = DiscordGuildEmoji.TryFromName(ctx.Client, name, out DiscordEmoji parsedEmoji);
                await ctx.EditResponseAsync(new DiscordWebhookBuilder().WithContent(parsedEmoji));
            }
            catch (NullReferenceException ex)
            {
                var failedMessage = new DiscordEmbedBuilder()
                {
                    Title = "Failed to get emoji!!!",
                    Description = "Error: " + ex.Message,
                    Color = DiscordColor.Red
                };
                await ctx.EditResponseAsync(new DiscordWebhookBuilder().AddEmbed(failedMessage));
            }
        }
```

So as you can see, what I'm currently doing is parsing in the emote as a parameter, the array helps to filter out its actual name since discord doesn't let you put an emote in the :emote: format, it just puts in the actual emoji. And I'm currently testing to see if it actually does return the emote back before I can convert it to a GIF/PNG

# Situations

The command works for any emotes from servers that the Bot is in, I'm assuming that's what the Client parameter is for so it can look through the guilds and search for this emoji

![image](https://user-images.githubusercontent.com/98812930/226107891-138e2e55-173e-4ff3-b783-035760e5295e.png)

However, for emotes from other servers, it fails to get the emoji
![image](https://user-images.githubusercontent.com/98812930/226107936-44f364d1-abbf-4061-841d-0c3fe4a8b008.png)

So is there any way I can get any emote that is parsed regardless of where the emote came from? Or is that just how the Method works, does the bot HAVE to be in the server for it to work?

**Comment by Giggitybyte** [comment, marked as answer]
You cannot use `DiscordGuildEmoji` for emotes which your bot cannot access. 

You should instead parse the content of the message to get the ID of each emote to [directly download](https://discord.com/developers/docs/reference#image-formatting) each image (e.g. `https://cdn.discordapp.com/emojis/786700993424392232.png` -> ![786700993424392232](https://cdn.discordapp.com/emojis/786700993424392232.png?size=24 ":pepe_lurk:")) then upload that image as a new emote on the current guild.

Naam has a similar feature in their ModCore project for the message context menu, if you'd like to use that as an example: https://github.com/Naamloos/ModCore/blob/master/ModCore/ContextMenu/MessageContextMenu.cs

**Reply by samjesus8** [reply]
Alright yeah, I thought that was the case where it couldn't access the server since its probably not in there
Ill do it this way then
